### PR TITLE
Add parseNFTokenID and tests

### PR DESCRIPTION
**Pull request by JST5000**
## High Level Overview of Change

Add a utility method to parse NFTokenIDs into a readable format.

### Context of Change

In response to #1897 

### Type of Change

- [X] New feature (non-breaking change which adds functionality)

## Test Plan

Added unit tests for this new feature.

### CODE REVIEW

**Comment by ihomp** [comment]
looks good! :) 
/ Bithomp

File: packages/xrpl/src/index.ts [hunk 1]
```diff
@@ -7,6 +7,8 @@ export * from './models'
 
 export * from './utils'
 
+export * from './sugar'
```
Comment (mvadari) [inline packages/xrpl/src/index.ts:10]: The `sugar` folder shouldn't be exported directly, it should only be exported via methods on the client.

**Review by mvadari — CHANGES_REQUESTED** [review]
See comments

File: packages/xrpl/src/utils/parseNFTokenID.ts [hunk 2]
```diff
@@ -0,0 +1,94 @@
+/* eslint-disable @typescript-eslint/no-magic-numbers -- Doing hex string parsing. */
+import BigNumber from 'bignumber.js'
+import { encodeAccountID } from 'ripple-address-codec'
+
+import { XrplError } from '../errors'
+
+/**
+ * All information encoded within an NFTokenID.
+ */
+interface NFTokenID {
```
Comment (mvadari) [inline packages/xrpl/src/utils/parseNFTokenID.ts:10]: Shouldn't this be in the `models` folder?

File: packages/xrpl/src/utils/parseNFTokenID.ts [hunk 3]
```diff
@@ -0,0 +1,94 @@
...
+ * |    |
+ * |    `---> TransferFee: 1337.0 bps or 13.37%
+ * |
+ * `---> Flags: 11 -> lsfBurnable, lsfOnlyXRP and lsfTransferable
+ *
+ * @param tokenID - A hex string which identifies an NFToken on the ledger.
+ * @throws XrplError when given an invalid tokenID.
+ * @returns a decoded tokenID with all information encoded within.
+ */
+function parseNFTokenID(tokenID: string): NFTokenID {
```
Comment (mvadari) [inline packages/xrpl/src/utils/parseNFTokenID.ts:72]: Are there plans to have a method that goes in the opposite direction?

Comment (JST5000) [reply packages/xrpl/src/utils/parseNFTokenID.ts:72]: I don't think the reverse matters, you'd need to know everything about the NFT which practically means you either minted the token, or just looked it up, and the minter can already get the NFTokenID I believe.

**Comment by JST5000** [comment]
@mvadari I'm still exporting NFTokenID from the parseNFTokenID file because you always need them together, does that make sense? (Or is it better to export from NFTokenID in the models folder?)

File: packages/xrpl/src/utils/parseNFTokenID.ts (hunk 3 above)
Comment (mvadari) [reply packages/xrpl/src/utils/parseNFTokenID.ts:55]: The grantees I've talked to made it sound like it's kind of a pain to dig out the NFTokenID from the depths of the transaction metadata - so if we could write the reverse function (or even just a method to do the digging of the metadata) I think that would also be super helpful. IIRC the only piece of information that you don't know at the time of sending the mint tx is the sequence #.

**Comment by mvadari** [comment]
> @mvadari I'm still exporting NFTokenID from the parseNFTokenID file because you always need them together, does that make sense? (Or is it better to export from NFTokenID in the models folder?)

I think in that case the way we've done it in this repo previously is to define the response of the method as the object instead of creating a new interface (e.g. https://js.xrpl.org/classes/Client.html#getBalances)

File: packages/xrpl/src/utils/parseNFTokenID.ts (hunk 3 above)
Comment (mvadari) [reply packages/xrpl/src/utils/parseNFTokenID.ts:55]: If you want to do that in a separate PR that's fine

**Comment by JST5000** [comment]
> I think in that case the way we've done it in this repo previously is to define the response of the method as the object instead of creating a new interface (e.g. https://js.xrpl.org/classes/Client.html#getBalances)

I think it's worth having it's own type because that allows for cleaner explanations for what each field represents. Especially if we're planning on making an "encode" function later on as well.

**Review by ckniffen — APPROVED** [review]

**Comment by mvadari** [comment]
> I think it's worth having it's own type because that allows for cleaner explanations for what each field represents. Especially if we're planning on making an "encode" function later on as well.

Hmm, I felt like it made it more confusing when there were a bunch of random types floating around instead of just an object.

But if we do want to keep the type, then, you're right, the type shouldn't go in `models`.

**Review by mvadari — APPROVED** [review]

**Pull request merged**